core/reactor.c

```diff
@@ -191,7 +191,7 @@ int _lf_do_step() {
             // container deadlines are defined in the container.
             // They can have different deadlines, so we have to check both.
             // Handle the local deadline first.
-            if (reaction->deadline > 0LL && physical_time > current_tag.time + reaction->deadline) {
+            if (physical_time > current_tag.time + reaction->deadline) {
```

Comment:
Not every reaction has a deadline. I suspect that the first check is there to make sure deadlines are only enforced for reactions that have one, but I'm not sure. The answer can be found in the code generator...

Reply:
But you're right that it is redundant given the check on line 185. As for the comment, I think it is outdated. We used to have different kinds of deadlines but discontinued other kinds we experimented with due to their complexity.

Reply:
Thank you for your response!
I will search about deadline in the code generator.
Thank you!

Reply:
@edwardalee Do you think you could take a look at the comments and let us if we need to update these comments? Maybe the two deadlines are not applicable anymore? Thanks!